File: 07-use-a-page-object/java/src/test/java/GoogleSearchPageObjTest.java
package src.test.java;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;

public class GoogleSearchPageObjTest {
    WebDriver driver;

    @Before
    public void setUp() throws Exception {
        driver = new FirefoxDriver();
    }

    @Test
    public void workWithBasicAuthTest() {
        GoogleSearchPageObj google = new GoogleSearchPageObj(driver);
        google.searchFor("elemental selenium tips");
        boolean result = google.searchResultPresent("Recieve a Free, Weekly tip");
        assert (result == true);
    }

    @After
    public void tearDown() throws Exception {
        driver.quit();
    }
}

File: 07-use-a-page-object/kotlin/src/test/kotlin/GoogleSearchPageObjTest.kt
import org.openqa.selenium.firefox.FirefoxDriver
import kotlin.test.AfterTest
import kotlin.test.BeforeTest
import kotlin.test.Test

class GoogleSearchPageObjTest {

    private val driver by lazy {
        FirefoxDriver()
    }

    @AfterTest
    fun tearDown() {
        // Quit session (closes browser)
        driver.quit()
    }

    @Test
    fun workWithBasicAuthTest() {
        val google = GoogleSearchPageObj(driver)
        google.searchFor("elemental selenium tips")
        val result = google.searchResultPresent("Receive a Free, Weekly Tip")
        assert(result)
    }

}